### PR TITLE
Deprecate beer-song and add bottle-song

### LINKED ISSUE #2505: Beer song: deprecate exercise and implement Bottle Song
As per https://github.com/exercism/problem-specifications/pull/2097, we should deprecate the `beer-song` exercise, and implement its replacement, [bottle-song](https://github.com/exercism/problem-specifications/tree/main/exercises/bottle-song).

## Contributing to this issue

Anyone is welcome to contribute to this issue. If you are interested in working on this, just post a comment below saying you want to work on this. Keep in mind that this task requires some exercism knowledge and might not be a good first issue if you are new to contributing.

Here are some docs related to this task:

- [Exercism Docs: Deprecating an exercise](https://exercism.org/docs/building/tracks/deprecated-exercises)
- [Exercism Docs: Practice Exercises](https://exercism.org/docs/building/tracks/practice-exercises)

### PR DESCRIPTION
**Pull request by PakkuDon**
Closes #2505 

As discussed in https://github.com/exercism/problem-specifications/issues/2066 we would like to replace the beer-song exercise with [bottle-song](https://github.com/exercism/problem-specifications/tree/main/exercises/bottle-song). This PR deprecates the beer-song exercise and adds a new bottle-song exercise

### CODE REVIEW

File: exercises/practice/bottle-song/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "authors": [],
```
Comment (andrerfcsantos) [inline exercises/practice/bottle-song/.meta/config.json:2]: Feel free to add yourself as the author of the exercise!

File: exercises/practice/bottle-song/bottle_song_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		})
+	}
+}
+
+func equal(a, b []string) bool {
+	if len(b) != len(a) {
+		return false
+	}
+
+	sort.Strings(a)
```
Comment (andrerfcsantos) [inline exercises/practice/bottle-song/bottle_song_test.go:25]: In this exercise the order of the verses matter, so we shouldn't order the verses to see that they are equal. Checking if the slices are equal as is should be enough. Also, in general we want to consider a nil slice equal to an empty non-nil slice, so if both slices have length 0, consider them equal. We have something similar in other exercises.

**Review by andrerfcsantos — CHANGES_REQUESTED** [review]
Thanks for this and for even making a generator!
Let some comments for now. In general the PR looks good, but I'll do a more through review later.

File: exercises/practice/bottle-song/bottle_song_test.go (hunk 2 above)
Comment (PakkuDon) [reply exercises/practice/bottle-song/bottle_song_test.go]: Good catch! I blindly copy-pasted this from another test 😅 
I've removed the sorts and added another if statement in

File: exercises/practice/bottle-song/.meta/example.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			fmt.Sprintf("%s green bottles hanging on the wall,", toTitleCase(numberToWord[n])),
+			"And if one green bottle should accidentally fall,",
+			fmt.Sprintf("There'll be %s green bottles hanging on the wall.", numberToWord[n-1]),
+		}
+	}
+}
+
+func toTitleCase(s string) string {
+	return strings.ToUpper(s[0:1]) + strings.ToLower(s[1:])
+}
```
Comment (PakkuDon) [inline exercises/practice/bottle-song/.meta/example.go:61]: I wasn't able to use `strings.Title` as it has been deprecated in Go 1.18 ([Build for reference](https://github.com/exercism/go/actions/runs/3250339595/jobs/5333890548))

Is there a better way to do this? The build suggests using `golang.org/x/text/cases` instead but that would involve installing an external dependency

Comment (andrerfcsantos) [reply exercises/practice/bottle-song/.meta/example.go:61]: Temporarily, feel free to copy the `strings.Title` and `strings.isSeparator` function directly: https://github.com/exercism/go/blob/43950ad7c2f1fc054f7487943f6f47a1228ac8af/exercises/practice/accumulate/accumulate_test.go#L10-L55

In the meantime, we are adding support to external packages to the track, and `golang.org/x/text` is one of the dependencies we'll support right out of the bat: https://github.com/exercism/go-test-runner/pull/84

When we have this, we'll replace all the copies of `strings.Title` we have.

Comment (PakkuDon) [reply exercises/practice/bottle-song/.meta/example.go]: Thanks! I've added that to `bottle_song_test.go` and updated the example solution

**Review by andrerfcsantos — APPROVED** [review]
Looks good!

Made some small adjustments to the configs. The difficulty of the exercise seems to not have changed, this was mostly a renaming, so I kept the difficulty for the old exercise. I also left the topics of the old exercise in the config for the new one. This is so that it helps our future work in converting the topics to the `practices` and `prerequisites` fields.

**Pull request merged**